## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 546521`
Commit: `089b84adf8423471f19a0207c211351ab472cbf8`

---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>brlcad</li>
    <li>starfox64recomp</li>
    <li>supercell-wx</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 25 packages built:</summary>
  <ul>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
    <li>eden</li>
    <li>gamescope</li>
    <li>gamescope-wsi</li>
    <li>geogram</li>
    <li>gimp (gimpPlugins.gimp)</li>
    <li>gimp-with-plugins</li>
    <li>gimpPlugins.gmic</li>
    <li>gimpPlugins.lightning</li>
    <li>gimpPlugins.resynthesizer</li>
    <li>hyperhdr</li>
    <li>imagelol</li>
    <li>imv</li>
    <li>lms</li>
    <li>nero-umu</li>
    <li>offpunk</li>
    <li>opengamepadui</li>
    <li>qoi</li>
    <li>shadps4</li>
    <li>shadps4-qtlauncher</li>
    <li>sm64tools</li>
    <li>stb</li>
    <li>timg</li>
    <li>xapp-thumbnailers</li>
  </ul>
</details>


---
<details>
<summary>brlcad</summary>
<pre>  asked CMake to find a package configuration file provided by &quot;TinyGLTF&quot;,
  but CMake did not find one.

  Could not find a package configuration file provided by &quot;TinyGLTF&quot; with any
  of the following names:

    TinyGLTF.cps
    tinygltf.cps
    TinyGLTFConfig.cmake
    tinygltf-config.cmake

  Add the installation prefix of &quot;TinyGLTF&quot; to CMAKE_PREFIX_PATH or set
  &quot;TinyGLTF_DIR&quot; to a directory containing one of the above files.  If
  &quot;TinyGLTF&quot; provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  src/libgcv/plugins/gltf/CMakeLists.txt:1 (BRLCAD_Find_Package)


-- Configuring incomplete, errors occurred!</pre>
</details>
<details>
<summary>supercell-wx</summary>
<pre>Skipping duplicate zone: VIZ001 St.Thomas...St. John.. and Adjacent Islands
Skipping duplicate zone: CAZ509 San Francisco Peninsula Coast
Processing states and territories file: /build/source/data/db/s_16ap26.dbf
Processing WFO file: /build/source/data/db/w_16ap26.dbf
[559/823] Building CXX object wxdata/CMakeFiles/wxdata.dir/source/scwx/wsr88d/rpg/vector_arrow_data_packet.cpp.o
[560/823] Building CXX object wxdata/CMakeFiles/wxdata.dir/source/scwx/wsr88d/rpg/wind_barb_data_packet.cpp.o
[561/823] Building CXX object wxdata/CMakeFiles/wxdata.dir/source/scwx/wsr88d/rpg/text_and_special_symbol_packet.cpp.o
[562/823] Building CXX object wxdata/CMakeFiles/wxdata.dir/source/scwx/zip/zip_stream_reader.cpp.o
[563/823] Building CXX object wxdata/CMakeFiles/wxdata.dir/source/scwx/zip/zip_stream_writer.cpp.o
FAILED: [code=1] wxdata/CMakeFiles/wxdata.dir/source/scwx/zip/zip_stream_writer.cpp.o 
/nix/store/3d1c302vw7kc8a5vknhmn34c0pd7zm6m-gcc-wrapper-15.3.0/bin/g++ -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_SHARED_LIB -I/build/source/external/hsluv-c/src -I/build/source/wxdata/include -I/build/source/wxdata/source -I/build/source/external/units/include -isystem /nix/store/0h5bizxank017bq7674x0z7qhgs54dzq-libxml2-2.15.3-dev/include/libxml2 -Wno-error=deprecated-declarations -Wno-error=free-nonheap-object -Wno-error=maybe-uninitialized -Wno-error=restrict -Wno-error=stringop-overflow -DQT_NO_USE_NODISCARD_FILE_OPEN -DBOOST_ALL_NO_LIB -O3 -DNDEBUG -std=c++20 -Wall -Wextra -Wpedantic -Werror -Wno-maybe-uninitialized -g -MD -MT wxdata/CMakeFiles/wxdata.dir/source/scwx/zip/zip_stream_writer.cpp.o -MF wxdata/CMakeFiles/wxdata.dir/source/scwx/zip/zip_stream_writer.cpp.o.d -o wxdata/CMakeFiles/wxdata.dir/source/scwx/zip/zip_stream_writer.cpp.o -c /build/source/wxdata/source/scwx/zip/zip_stream_writer.cpp
/build/source/wxdata/source/scwx/zip/zip_stream_writer.cpp: In member function &#x27;bool scwx::zip::ZipStreamWriter::Impl::AddFile(const std::string&amp;, const T&amp;, zip_flags_t)&#x27;:
/build/source/wxdata/source/scwx/zip/zip_stream_writer.cpp:370:9: error: &#x27;memcpy&#x27; is not a member of &#x27;std&#x27;; did you mean &#x27;wmemcpy&#x27;? [-Wtemplate-body]
  370 |    std::memcpy(buffer, content.data(), content.size());
      |         ^~~~~~
      |         wmemcpy
[564/823] Building CXX object wxdata/CMakeFiles/wxdata.dir/source/scwx/wsr88d/rpg/storm_tracking_information_message.cpp.o
[565/823] Building CXX object external/date/CMakeFiles/date-tz.dir/src/tz.cpp.o
[566/823] Automatic MOC and UIC for target scwx-qt
ninja: build stopped: subcommand failed.</pre>
</details>

--- a/pkgs/by-name/_2/_2ship2harkinian/package.nix
+++ b/pkgs/by-name/_2/_2ship2harkinian/package.nix
@@ -19,6 +19,7 @@
   libpulseaudio,
   libzip,
   nlohmann_json,
+  stb,
   SDL2,
   spdlog,
   tinyxml-2,
@@ -80,12 +81,6 @@ let
       #define STB_IMAGE_IMPLEMENTATION
       #include "stb_image.h"
     '';
-  };
-
-  stb' = fetchurl {
-    name = "stb_image.h";
-    url = "https://raw.githubusercontent.com/nothings/stb/0bc88af4de5fb022db643c2d8e549a0927749354/stb_image.h";
-    hash = "sha256-xUsVponmofMsdeLsI6+kQuPg436JS3PBl00IZ5sg3Vw=";
   };
 
   stormlib' = applyPatches {
@@ -213,7 +208,7 @@ stdenv.mkDerivation (finalAttrs: {
   preConfigure = ''
     # mirror 2ship's stb
     mkdir stb
-    cp ${stb'} ./stb/${stb'.name}
+    cp ${stb}/include/stb/stb_image.h ./stb/stb_image.h
     cp ${stb_impl} ./stb/${stb_impl.name}
     substituteInPlace libultraship/cmake/dependencies/common.cmake \
       --replace-fail "\''${STB_DIR}" "$(readlink -f ./stb)"

--- a/pkgs/by-name/no/noctalia-greeter/package.nix
+++ b/pkgs/by-name/no/noctalia-greeter/package.nix
@@ -28,18 +28,6 @@
   nix-update-script,
 }:
 
-let
-  # nixpkgs stb doesn't have stb_image_resize2.h which noctalia-greeter needs
-  stb' = stb.overrideAttrs {
-    version = "0-unstable-2025-10-26";
-    src = fetchFromGitHub {
-      owner = "nothings";
-      repo = "stb";
-      rev = "f1c79c02822848a9bed4315b12c8c8f3761e1296";
-      hash = "sha256-BlyXJtAI7WqXCTT3ylww8zoG0hBxaojJnQDvdQOXJPE=";
-    };
-  };
-in
 stdenv.mkDerivation (finalAttrs: {
   pname = "noctalia-greeter";
   version = "1.2.1";
@@ -73,7 +61,7 @@ stdenv.mkDerivation (finalAttrs: {
     libxkbcommon
     nlohmann_json
     pango
-    stb'
+    stb
     tomlplusplus
     wayland
     wayland-protocols

--- a/pkgs/by-name/no/noctalia/package.nix
+++ b/pkgs/by-name/no/noctalia/package.nix
@@ -49,18 +49,6 @@
   gitMinimal,
 }:
 
-let
-  # nixpkgs stb doesn't have stb_image_resize2.h which noctalia needs
-  stb' = stb.overrideAttrs {
-    version = "0-unstable-2025-10-26";
-    src = fetchFromGitHub {
-      owner = "nothings";
-      repo = "stb";
-      rev = "f1c79c02822848a9bed4315b12c8c8f3761e1296";
-      hash = "sha256-BlyXJtAI7WqXCTT3ylww8zoG0hBxaojJnQDvdQOXJPE=";
-    };
-  };
-in
 stdenv.mkDerivation (finalAttrs: {
   __structuredAttrs = true;
 
@@ -111,7 +99,7 @@ stdenv.mkDerivation (finalAttrs: {
     pipewire
     polkit
     sdbus-cpp_2
-    stb'
+    stb
     systemdLibs
     tomlplusplus
     wayland

--- a/pkgs/by-name/sh/shipwright/package.nix
+++ b/pkgs/by-name/sh/shipwright/package.nix
@@ -25,6 +25,7 @@
   nlohmann_json,
   tinyxml-2,
   spdlog,
+  stb,
   writeTextFile,
   fixDarwinDylibNames,
   applyPatches,
@@ -81,12 +82,6 @@ let
       #define STB_IMAGE_IMPLEMENTATION
       #include "stb_image.h"
     '';
-  };
-
-  stb' = fetchurl {
-    name = "stb_image.h";
-    url = "https://raw.githubusercontent.com/nothings/stb/0bc88af4de5fb022db643c2d8e549a0927749354/stb_image.h";
-    hash = "sha256-xUsVponmofMsdeLsI6+kQuPg436JS3PBl00IZ5sg3Vw=";
   };
 
   stormlib' = applyPatches {
@@ -208,7 +203,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   preConfigure = ''
     mkdir stb
-    cp ${stb'} ./stb/${stb'.name}
+    cp ${stb}/include/stb/stb_image.h ./stb/stb_image.h
     cp ${stb_impl} ./stb/${stb_impl.name}
     substituteInPlace libultraship/cmake/dependencies/common.cmake \
       --replace-fail "\''${STB_DIR}" "$(readlink -f ./stb)"

--- a/pkgs/by-name/sp/spaghettikart/package.nix
+++ b/pkgs/by-name/sp/spaghettikart/package.nix
@@ -24,6 +24,7 @@
   SDL2,
   SDL2_net,
   spdlog,
+  stb,
   tinyxml-2,
   tomlplusplus,
   zenity,
@@ -81,12 +82,6 @@ let
       #define STB_IMAGE_IMPLEMENTATION
       #include "stb_image.h"
     '';
-  };
-
-  stb' = fetchurl {
-    name = "stb_image.h";
-    url = "https://raw.githubusercontent.com/nothings/stb/0bc88af4de5fb022db643c2d8e549a0927749354/stb_image.h";
-    hash = "sha256-xUsVponmofMsdeLsI6+kQuPg436JS3PBl00IZ5sg3Vw=";
   };
 
   stormlib' = applyPatches {
@@ -219,7 +214,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   preConfigure = ''
     mkdir stb
-    cp ${stb'} ./stb/${stb'.name}
+    cp ${stb}/include/stb/stb_image.h ./stb/stb_image.h
     cp ${stb_impl} ./stb/${stb_impl.name}
     substituteInPlace libultraship/cmake/dependencies/common.cmake \
       --replace-fail "\''${STB_DIR}" "$(readlink -f ./stb)"

--- a/pkgs/by-name/st/starship-sf64/package.nix
+++ b/pkgs/by-name/st/starship-sf64/package.nix
@@ -24,6 +24,7 @@
   nlohmann_json,
   SDL2,
   spdlog,
+  stb,
   tinyxml-2,
   zenity,
   sdl_gamecontrollerdb,
@@ -74,12 +75,6 @@ let
       #define STB_IMAGE_IMPLEMENTATION
       #include "stb_image.h"
     '';
-  };
-
-  stb' = fetchurl {
-    name = "stb_image.h";
-    url = "https://raw.githubusercontent.com/nothings/stb/0bc88af4de5fb022db643c2d8e549a0927749354/stb_image.h";
-    hash = "sha256-xUsVponmofMsdeLsI6+kQuPg436JS3PBl00IZ5sg3Vw=";
   };
 
   stormlib' = applyPatches {
@@ -195,7 +190,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   preConfigure = ''
     mkdir stb
-    cp ${stb'} ./stb/${stb'.name}
+    cp ${stb}/include/stb/stb_image.h ./stb/stb_image.h
     cp ${stb_impl} ./stb/${stb_impl.name}
     substituteInPlace libultraship/cmake/dependencies/common.cmake \
       --replace-fail "\''${STB_DIR}" "$(readlink -f ./stb)"

--- a/pkgs/by-name/st/stb/package.nix
+++ b/pkgs/by-name/st/stb/package.nix
@@ -4,17 +4,18 @@
   fetchFromGitHub,
   copyPkgconfigItems,
   makePkgconfigItem,
+  unstableGitUpdater,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "stb";
-  version = "0-unstable-2023-01-29";
+  version = "0-unstable-2026-04-15";
 
   src = fetchFromGitHub {
     owner = "nothings";
     repo = "stb";
-    rev = "5736b15f7ea0ffb08dd38af21067c314d6a3aae9";
-    hash = "sha256-s2ASdlT3bBNrqvwfhhN6skjbmyEnUgvNOrvhgUSRj98=";
+    rev = "31c1ad37456438565541f4919958214b6e762fb4";
+    hash = "sha256-m2yNUlA37hDkKQVrQ+R8nufHfW/cXLnMo+n1X1Cyun0=";
   };
 
   nativeBuildInputs = [ copyPkgconfigItems ];
@@ -41,6 +42,8 @@ stdenv.mkDerivation (finalAttrs: {
     cp *.c $out/include/stb/
     runHook postInstall
   '';
+
+  passthru.updateScript = unstableGitUpdater { };
 
   meta = {
     description = "Single-file public domain libraries for C/C++";

